python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -520,10 +525,40 @@ def compute_result_column_index(self):
             else:
                 return aggs_as_list
         else:
-            if len(aggs_as_list) == len(self.aggs):
+            return_multi_index = True
+            if isinstance(self.original_aggs, str):
+                return_multi_index = False
+            if isinstance(self.original_aggs, collections.abc.Mapping):
+                return_multi_index = False
+                for key in self.original_aggs:
+                    if not isinstance(self.original_aggs[key], str):
+                        return_multi_index = True
+                        break
+            if return_multi_index:
+                return MultiIndex.from_tuples(zip(value_names, aggs_as_list))
+            else:
                 return value_names
+            """
```

Comment:
Is the commented code being left here on purpose?

Reply:
Nope 😄